core/ledger/ledger.go

```diff
@@ -84,8 +84,8 @@ func (self *Ledger) GetStateMerkleRoot(height uint32) (result common.Uint256, er
 	return self.ldgStore.GetStateMerkleRoot(height)
 }
 
-func (self *Ledger) GetBlockRootWithNewTxRoot(txRoot common.Uint256) common.Uint256 {
-	return self.ldgStore.GetBlockRootWithNewTxRoot(txRoot)
+func (self *Ledger) GetBlockRootWithNewTxRoots(txRoots []common.Uint256) common.Uint256 {
```

Comment:
suggest changing this interface to
GetBlockRootWithNewTxRoots(txRootHeight uint32, newTxRoots []common.Uint256)
： newTxRoots includes merkle roots starting from block txRootHeight

when calculating the new root, ledger should check if any txRoots missed or inconsistent.